### DISCUSSION TITLE
Custom Routing using react-router-dom

### DISCUSSION THREAD

**Discussion by karlamarice** [vikejs/vike#1177] [Discussion] [answered]
Hello, I'm currently working on migrating an older project from vite-plugin-ssr to Vike. I'm wondering if it's feasible to customize the routing using react-router-dom in this context. Additionally, I'm looking for documentation or guidance on the migration process. The issue I'm facing at the moment is related to the routing, specifically due to the usage of BrowserRouter, Route, and Routes from react-router-dom in my project.

```
[vike][request(3)] Following error was thrown by the render() hook defined at /src/renderer/_default.page.server.tsx > `export { render }`
Error: My render() hook expects pageContext.Page to be defined
    at ....
```

**Comment by brillout** [comment, marked as answer]
Have you seen https://vike.dev/vue-router-and-react-router? What's your motivation for using React Router?

> Additionally, I'm looking for documentation or guidance on the migration process

Have you seen https://vite-plugin-ssr.com/vike?

> [vike][request(3)] Following error was thrown by the render() hook defined at /src/renderer/_default.page.server.tsx > `export { render }`
> Error: My render() hook expects pageContext.Page to be defined

That's a user land issue.

**Comment by karlamarice** [comment]
Hey! I've just completed the migration of my application to vike. I managed to resolve my custom routing issue using this resource: https://vike.dev/route-string. It proved to be simpler than using react-router-dom. Now, I'm eagerly looking forward to incorporating more functionalities into our application. Thank you so much!

**Reply by brillout** [reply]
👍

(Btw. in case your company is up for it: https://github.com/sponsors/brillout.)
